File: Assets/zfoocs/Websocket/WebSocketObjectB.cs
using System;
using System.Collections.Generic;
namespace zfoocs
{
    
    public class WebSocketObjectB
    {
        public bool flag;
    }

    public class WebSocketObjectBRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 2072;
        }
    
        public void Write(ByteBuffer buffer, object packet)
        {
            if (packet == null)
            {
                buffer.WriteInt(0);
                return;
            }
            WebSocketObjectB message = (WebSocketObjectB) packet;
            buffer.WriteInt(-1);
            buffer.WriteBool(message.flag);
        }
    
        public object Read(ByteBuffer buffer)
        {
            int length = buffer.ReadInt();
            if (length == 0)
            {
                return null;
            }
            int beforeReadIndex = buffer.GetReadOffset();
            WebSocketObjectB packet = new WebSocketObjectB();
            bool result0 = buffer.ReadBool();
            packet.flag = result0;
            if (length > 0)
            {
                buffer.SetReadOffset(beforeReadIndex + length);
            }
            return packet;
        }
    }
}

File: Assets/zfoolua/websocket/WebSocketObjectB.lua

local WebSocketObjectB = {}

function WebSocketObjectB:new()
    local obj = {
        flag = false -- bool
    }
    setmetatable(obj, self)
    self.__index = self
    return obj
end

function WebSocketObjectB:protocolId()
    return 2072
end

function WebSocketObjectB:write(buffer, packet)
    if packet == nil then
        buffer:writeInt(0)
        return
    end
    buffer:writeInt(-1)
    buffer:writeBoolean(packet.flag)
end

function WebSocketObjectB:read(buffer)
    local length = buffer:readInt()
    if length == 0 then
        return nil
    end
    local beforeReadIndex = buffer:getReadOffset()
    local packet = WebSocketObjectB:new()
    local result0 = buffer:readBoolean()
    packet.flag = result0
    if length > 0 then
        buffer:setReadOffset(beforeReadIndex + length)
    end
    return packet
end

return WebSocketObjectB
